### PR TITLE
Arconomy: Amputating the Invisible Hand of Income.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
Version 1.5.

**Hey, hold up! This PR should be merged alongside #54715! Otherwise the experience will be miserable!**

## About The Pull Request

I hit all the fun economy goals except for this one because it's a straight up nerf and it's pretty massive, but it's necessary to add stakes to the existing systems.

### Player income.
Passive income is dead. Outside of roundstart income and civilian bounties, players will no longer make income periodically throughout the round. It's pretty clear that players make too much money, and while I'm happy with player's roundstart ratio's of credits, the longer a round goes on, the less value money has to the vast majority of players. If you can afford something roundstart and purchase it then, that's nice, but if not you can always just wait it out, nullifying engagement with a system that has the potential to drive gameplay and roleplay.

#### Very Brief FAQ on the above if you think it's horrible.
I've gotten quite a few questions about why I believe this is the correct direction to take this system, so let's take it from the top. You're playing a station engineer, and start the shift with 525 credits (Actually 600 because you get an automatic paycheck moments afterwards but what's a few dozen credits between friends). Your goal is to purchase an RCD. An RCD costs 1700 credits at roundstart, and as a premium item, is not subject to inflation. Your goal is to buy one as soon as possible. So, perhaps you could sell services to different departments and make a few hundred credits to get a really early RCD. Perhaps you could do a civilian bounty or two. You could even perform some kind of complicated money laundering scheme between pricetags, stealing the CE's tablet for departmental purchases, or some other interactive way to engage with the round while interacting with money.
... Or you could wait a few minutes and buy one anyway. There are several things in-game that are balanced purely by time locked interactions, and on many of those subjects I fully understand why. RnD is locked behind time because there's an active progression on what value those objects give you. Tier 2 cells hold more power than tier 1, 3 than 2, what have you. That said, many of the things locked behind money alone in Space Station 13 don't hold that same clean sense of progression. 95% of the objects purchased with money alone are either unique objects unobtainable through other means, or ARE obtained typically by a timelocked system, and are imposed over an extreme cost. By the time that you would passively have made that money in a single round, almost 2 hours would have gone by if you had zero other interaction with the in-game economy, and that's assuming you neither buy anything nor somehow gain any other money. 

That comes to 2 major conclusions: 
The money you obtained passively doesn't give a player any real attachment to it, because there was zero effort or though placed in how it was acquired. 
Things that can obtained with large amounts of money aren't even considered as an avenue to obtain them if they can be obtained otherwise, because everything can be obtained, eventually, even if you never touch your ID once.

Removing passive income will place a higher individual value on the money players have, provide a real reason to interact with players or offer money for the purpose of spending cash, and enforce the idea that losing money has a real consequence, all at once

### Departmental Income.
Departmental income has additionally been... well I would say "rebalanced" but lets just go with "balanced for the first time" to cover the fact that there will no longer be any passive drain on  those accounts throughout the shift.

**This section used to be a lot longer but after discussion with maintainers has been heavily adjusted in a different direction.**
Instead of departmental budgets getting wild, sweeping formulas to make keeping track of how each department's making colossal amounts of money each shift, every department has been moved to the cargo income standard, where they're making a static 500 credits every 5 minutes. While this is certainly a bit less engaging, it also keeps the amount of supply and power of individual departments from getting too out of control.

### Changes to Costs and Prices
Due to the sweeping nature of income and the value of money this brings, there are several aspects of this that are worth mentioning:
**Vendors that previously sold job-specific objects now offer those things at a steep discount instead.** Specifically, 80%. Less ways to bypass existing cost balancing on station, while simultaneously keeping it cheap and accessible for people in that department.
**Several extremely expensive objects have had their prices lowered to match the limited amounts of money available.** This is in regard to more of the higher end objects that serve a unique purpose as opposed to things that are just available  much much later in the game. For example, amputation shears have been lowered in price, but, RCDs have not.

## Why It's Good For The Game

I said I was going to kill passive income from the very beginning, I've done a lot to mitigate the amount of harm income has on the rest of the station, and made it so having a low playing job has a balanced, autocorrecting mechanism enabling you to earn more money if you need it in-game. That said, it won't be possible until passive income is gone. By round end average crew wealth is in the thousands of credits, as noted  by the average Bagil round combined wealth sits in the 10s of thousands on most any shift, unless the shift ends in a matter of minutes instead.

What this PR Enables with it's inclusion:
  * Players are tasked with making serious choices about when and how they spend their money. If all it takes is another 10 minutes to buy something,  that means it's either not worth buying if it's not available now, or it's worth waiting for and not worth going out of your way to acquire the funds as part of your round.
  * Department funds are now on par with a reasonable slow build up of resources over time. Only on stations where there are enough players to warrant giving it are departments making enough money to seriously use their funds to purchase goods/pay out bounties, based on the scaling that individual impact provides now.
  * Fills out the original economy design doc to finally put in place the system of scarcity with it's desired consequences on the whole.

## Changelog
:cl:
del: Player passive income has been removed.
balance: Departmental budgets income has been thoroughly reworked, as they no longer need to balance the whole of the crew's income off of them, making them a more reasonable resource pool scaled to the station's size or efforts.
balance: Departmental income is a static value for every department, getting more funds every 5 minutes (Plus whatever's added from vending machines!)
balance: Buying goods through your department's vendors now provides an 80% discount as opposed to being completely free.
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
does sec department get money from gulag reclaimer? ( it doesnt generate miner points which can be converted to cash)

**Comment by SkeletalElite** [comment]
> does sec department get money from gulag reclaimer? ( it doesnt generate miner points which can be converted to cash)

yeah sec budget should get money from gulagging people.

**Comment by Winterous** [comment]
So, the individual's wealth is supposed to only grow by completing bounties now? Or is there a way to pay wages through the department's budget that simply isn't automatic?

And, I feel like the system as it is doesn't make much sense, nor would this new one...
The station is predominately a mining facility with research activity going on, right? If so, its income should be almost entirely from exports, with some financial support for research efforts.

If that's the case, wouldn't it make most sense for there to be a slow trickle of that grant money, and simply have all cargo exports be divided up between departments, based upon need?
Individuals not earning a salary I can appreciate, your reward is the life you lead, the support structures in place; your job is the cost of citizenship in this corporate state. But then, it does strike as a little strange that there is any personal cost to things, theoretically you should be paying directly from your department budget, needing to get permission, or have a limited expense account.

It just feels... Off, you know?

**Comment by Barhandar** [comment]
So basically, most people can no longer afford anything because they're BUSY DOING THEIR JOB, and the few who _can_ go do bounties (or steal other people's cash, all praise Murrican Medbay) are rich just like they are now.

Horrible, terrible, no good, very bad idea. People **should not need** to do things irrelevant to vast majority of the jobs (there is exactly 1 - ONE - job for which bounties are part of description, and it's the cargo technician) to be able to buy the things IN THEIR DEPARTMENT.  
Until income is tied directly into "doing your job", there needs to be passive income. And because for many jobs you CANNOT make acceptable algorithm for detecting whether it's being done (good luck detecting codewise that detective, engineer, atmostech, etc are actually working and not just gaming the system), passive income needs to stay.

**Comment by Mickyan** [comment]
I think the new formulas for crew budgets look good, with the way inflation works right now I don't think personal passive income is an issue, if I have to buy something moderately expensive I will likely have to do a few bounties and it will be a long while before I can buy anything else. As it's been mentioned, rewarding people for doing bounties instead of their actual job is not a particularly good alternative

Even though it's out of scope, ideally heads of staff should be able to allocate lower/higher payouts, or just automate payouts to scale with the budget grants (we tried this before but it didn't work because the formulas sucked), so that people are encouraged to perform well within their department, instead of just on a personal level

**Comment by RaveRadbury** [comment]
I look forward to seeing this live and trying it out.

**Comment by Winterous** [comment]
I think that a departmental expense account is a good way of compromising. You essentially have a budget you can spend from your department, with a limit; you either spend that money or you don't get more money to spend.

It's like how various businesses will buy a shitload of chairs or computers or whatever because they still have budget, and if they don't spend it, then it goes to waste because their budget's gonna get refreshed in a few weeks anyway. It fits with the corporate setting nicely, I think.

**Comment by ExcessiveUseOfCobblestone** [comment]
is there anything practical that requires passive income on dept budget?

Money is never going to be in demand if you have a 100% braindead way for it to be passively and continually supplied into the round. How you are suppose to balance against say lowpop and highpop for medical? How are you suppose to balance money when you have things like SM count trying to be comparable to say 25 rare slimes? I can't.

**Comment by actioninja** [comment]
wtf assistant nerf

**Comment by bluetshirtguy** [comment]
I'm very worried how this will effect botany. otherwise, meh seems okay.

**Comment by anconfuzedrock** [comment]
so either do bounties (aka make the station worse and waste your own time for money. I don't see the benefit in a sec officer grabbing every prisoner jumpsuit and selling them for 200 bucks) instead of your job, or spend all of your money roundstart before it's less valuable?

**Comment by Jared-Fogle** [comment]
To be clear, everything Arcane said in the OP is *already a thing*, most likely you just didn't know about it. The only thing this PR does is remove passive income.

Saying this because it confused me.

**Comment by djcart137** [comment]
From the standpoint of a sec/lawyer main:
- Nobody can afford the security webbing item roundstart. I don't think you will ever see anybody buy it from now on, even though it looks really really cool (and was sort of a trademark of my HoS character).
- This really messes with citations as an RP mechanic, as any serious ones require the offending player to stop doing their job and start doing bounties.
- This really messes with lawyers seeking monetary compensation for clients against individuals in civil court, and charging private clients for representation.
- There is no new potential to drive gameplay or roleplay as a sec main when you have to stop doing your job to make money, unless you're implying that sec mains should start taking bribes from players who do bounties in order to avoid arrests. The value produced by security is not material, and cannot be reflected by a bounty system.
- There is now no in-universe reason to be a scientist, engineer, or security officer instead of an assistant, if everyone has access to the same amenities regardless of their position, qualifications, or labor. Wages placed a value on skilled labor.
- Assistants can now easily be the richest players, which is the opposite of what should happen.

Maybe I misunderstand the system. I think nerfing passive income is a great idea. I don't think punishing players for doing their job instead of doing bounties is.

**Comment by Winterous** [comment]
Maybe remove passive income for assistants only (or a few other things like lawyer, curator, and chaplain), so that they actually have to ASSIST departments to earn an income, or else ship out random garbage for bounties.

**Comment by Dorsisdwarf** [comment]
> 
> 
> To be clear, everything Arcane said in the OP is _already a thing_, most likely you just didn't know about it. The only thing this PR does is remove passive income.
> 
> Saying this because it confused me.

Yeah, but now instead of an amusing little sidegame for bored assistants who need cash to buy a more expensive skateboard it's the only way anyone can buy tools, drinks, etc. without having to beg their head of staff to give them pocket money.

**Comment by Mickyan** [comment]
To make a concrete example I was playing as paramedic earlier and at the 1 HOUR mark (most shifts outside of manuel are already over at this point) I was still short of about 300 credits to buy a pinpointer despite having saved all of my money throughout the shift, so I did a couple of bounties in my free time and bought the damn thing.

To me that already seems like what Arcane is trying to do but without having the entire system be dependent on bounties, which are typically very different from doing your actual job in a way that benefits the station. 
I believe this PR is a good step in making the money intake from department grants more sane and I could even see reasons for further nerfs to individual payouts, but the removal of passive income seems like barking up the wrong tree.

**Comment by Jared-Fogle** [comment]
@Dorsisdwarf Yeah, I just wanted to make sure that was clear to everyone else, since it wasn't to me.

**Comment by ArcaneMusic** [comment]
> is there anything practical that requires passive income on dept budget?
> 
> Money is never going to be in demand if you have a 100% braindead way for it to be passively and continually supplied into the round. How you are suppose to balance against say lowpop and highpop for medical? How are you suppose to balance money when you have things like SM count trying to be comparable to say 25 rare slimes? I can't.

Passive income for department budget is built in in order to still slow departmental ordering from heads of staff in their relative department, and allows for heads of staff to have greater interaction with obtaining supplies necessary for their department.
Relevant PR: #53881

**Comment by SkeletalElite** [comment]
I think it makes sense to remove passive income for some roles, but for most jobs making money without passive income means literally ignoring your job.
Jobs like assisstant? Sure why not. Security? Guess they'll literally never buy sec webbings anymore, they already pretty much didn't.

**Comment by JMoldy** [comment]
And what happens if the head of your department isn't available roundstart? Is this just a massive fuck-you to any low-pop ever, guaranteeing that over half the crew is going to be unable to buy effectively anything at all?

**Comment by Tlaltecuhtli** [comment]
on low pop everyone has aa anyway

**Comment by interestingusernam3** [comment]
>Assistants can now easily be the richest players, which is the opposite of what should happen.

No this is exactly what should happen

**Comment by cacogen** [comment]
I don't know if the credits in the departmental budgets can be used interchangeably with personal credits but the new ways that departments generate credits passively seems misguided and easily exploited. I understand how difficult it is to measure productivity in this game which is why basing income off any attempt to seems fruitless. Better to allow players to control wages to reward productivity while limiting total credits available to prevent exploitation. 

While I'm happy scarcity is finally in the cards I think removing passive income from players just punishes people for doing their jobs instead of bounties. Instead, it would be better if the station had a limited amount of credits and each head chose the wages of each of their employees to reward or punish people based on their productivity. 

The problem with making currency that players actually value is nobody is playing jobs because they need money. They're working because they want items or abilities gated by their jobs or because the gameplay is enjoyable. Money is not needed to survive so it remains optional and not very useful. There aren't enough cool things or gameplay advantages gated by money so it doesn't have enough uses to be valuable. The things that people want are unlocked by jobs or limited to certain roles. I can't buy a pulse rifle, for instance.

**Comment by ExcessiveUseOfCobblestone** [comment]
> 
> 
> > is there anything practical that requires passive income on dept budget?
> > Money is never going to be in demand if you have a 100% braindead way for it to be passively and continually supplied into the round. How you are suppose to balance against say lowpop and highpop for medical? How are you suppose to balance money when you have things like SM count trying to be comparable to say 25 rare slimes? I can't.
> 
> Passive income for department budget is built in in order to still slow departmental ordering from heads of staff in their relative department, and allows for heads of staff to have greater interaction with obtaining supplies necessary for their department.
> Relevant PR: #53881

make it a set amount so im not trying to compare amount of slimes to health of crew to amount of SMs.

**Comment by ArcaneMusic** [comment]
Discussed things on this going forward:

- [ ] Look at the most expensive on-station objects and bring them inline with the more reasonable station wealth going forward.
- [ ] Leave open the designspace for possible ways to reward the station with cash uniformly, but without reverting to just  passive income. (Event that might reward the station with a paycheck advance for example, weighed against consequences.)
- [ ] Determine common goals between the two econ camps (Ask me about this on discord!)

File: code/controllers/subsystem/economy.dm [hunk 1]
```diff
@@ -45,8 +45,17 @@ SUBSYSTEM_DEF(economy)
 							"oil" = 75,
 							"adamantine" = 75,
 							// tier 4
+<<<<<<< HEAD
```
Comment (Jared-Fogle) [inline code/controllers/subsystem/economy.dm]: lol

**Comment by Doosteroni** [comment]
This blows

**Comment by Maurukas** [comment]
This was testmerged on Bagil in round ID 149150 and it's preceding round.

The first issue I ahelped to you, which was that that free items in the nanomed vendor were charging for these items.  In the ahelp, you indicated that they were supposed to only charge 10% of the normal price. Thank you for the refund, by the way.

In 149150, I noticed at round start, that synthflesh gel, a premium item normally 500cr, was inflated to 920cr. I believe there may be an issue with inflation with these changes. My starting 400cr got two items out of the nanomed before being exhausted. The only opportunity I had for more money during this shift was looting a dead geneticist's card. At another point, every member of medbay was broke, and we could not get bone gel out of the nanomed. We had to go fetch the CMO from the bridge and have him pay for it. Several other staff members from other departments commented that they were unable to afford their basic round start equipment.

I think that the costs of the items in round start needs to be looked at in the context of these changes, and adjusted downward by a significant amount.

**Comment by brotherbeyondd** [comment]
All "free" items can't be bought during the testmerges on Bagil.
A key example of this are the janitor's vending machine where a lot of the items are free for the jannie.
Now jannies just complain that they cant buy their two soaps roundstart, please fix.

**Comment by saprasam** [comment]
free items arent free, fix your code pretty please

**Comment by saprasam** [comment]
i also have to point out that one of the main ways of getting money (Civillian Bounty Console) was completely revamped and now doesn't give you any money, only a bounty cube for cargo
this clearly wasn't testmerged with any consideration for the game right now, you cant get money unless you're on delta, which has 1000 quid lying in maints.
Thanks Coders !

**Comment by Winterous** [comment]
> 
> 
> i also have to point out that one of the main ways of getting money (Civillian Bounty Console) was completely revamped and now doesn't give you any money, only a bounty cube for cargo
> this clearly wasn't testmerged with any consideration for the game right now, you cant get money unless you're on delta, which has 1000 quid lying in maints.
> Thanks Coders !

You get money once the bounty cube is shipped.

**Comment by saprasam** [comment]
> You get money once the bounty cube is shipped.

that's stupid

**Comment by Winterous** [comment]
Well, cargo ALSO gets money from shipping it, so it's not like they'll never do it.

**Comment by ArcaneMusic** [comment]
> 
> 
> free items arent free, fix your code pretty please

If you read the PR body, which you clearly didn't, you'll see that part of the experimental changes for this PR was to change job specific free items from a 100% discount to an 80% discount.
I pulled the testmerge off bagil because I have several fixes and changes I need first before I'm comfortable testmerging this again.
Plus, the whole limiting the cash available in-round into a limited currency was the primary goal of the PR.

**Comment by SkeletalElite** [comment]
Basic department stuff not being free is literally terrible.

**Comment by ArcaneMusic** [comment]
> Basic department stuff not being free is literally terrible.

The above linked PR (**54715**) is being done in an attempt to make it barable, but it was made as a seperate PR as an atomized change to keep this PR distinct in it's goals.

File: tgui/packages/tgui/interfaces/Vending.js [hunk 2]
```diff
@@ -22,9 +22,12 @@ const VendingRow = (props, context) => {
       !product.premium
       && department
       && user
-      && department === user.department
     )
   );
+  const discount = (
+    department === user.department
+  );
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/Vending.js:30]:
```suggestion
  const discount = department === user.department;
```

File: tgui/packages/tgui/interfaces/Vending.js [hunk 3]
```diff
@@ -79,7 +82,7 @@ const VendingRow = (props, context) => {
                 || product.price > data.user.cash
               )
             )}
-            content={free ? 'FREE' : product.price + ' cr'}
+            content={(free && discount) ? redPrice +' cr': product.price+' cr'}
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/Vending.js:86]:
```suggestion
            content={(free && discount) ? `${redPrice} cr` : `${product.price} cr`}
```

**Comment by SkeletalElite** [comment]
TFW you're playing sec but literally can't even buy zipties to arrest people because you're too busy stopping shitters and antags to collect prison uniforms

**Review by stylemistake — DISMISSED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
ill take a look at this soon ™️

**Comment by basedcringe** [comment]
This is just fucking stupid do you even play the game

File: code/controllers/subsystem/economy.dm [hunk 4]
```diff
@@ -15,36 +15,37 @@ SUBSYSTEM_DEF(economy)
...
 							//Take care when enabling, as players will NOT respond well if the economy is set up for low cash flows.
-	var/alive_humans_bounty = 100
+	var/alive_humans_bounty = 50
 	var/crew_safety_bounty = 1500
 	var/monster_bounty = 150
 	var/mood_bounty = 100
 	var/techweb_bounty = 250
-	var/slime_bounty = list("grey" = 10,
+	var/civ_bounty_value = 25
+	var/slime_bounty = list("grey" = 1,
```
Comment (ExcessiveUseOfCobblestone) [inline code/controllers/subsystem/economy.dm:24]: does any of this have use after this pr

File: code/controllers/subsystem/economy.dm [hunk 5]
```diff
@@ -15,36 +15,37 @@ SUBSYSTEM_DEF(economy)
...
+							"cerulean" = 75,
+							"sepia" = 75,
+							"bluespace" = 75,
+							"pyrite" = 75,
+							"light pink" = 75,
+							"oil" = 75,
+							"adamantine" = 75,
 							// tier 4
-							"rainbow" = 1000)
+							"rainbow" = 100)
```
Comment (ExcessiveUseOfCobblestone) [inline code/controllers/subsystem/economy.dm:48]: wow its almost like these values should be defines 😠

File: code/controllers/subsystem/economy.dm [hunk 6]
```diff
@@ -91,79 +87,40 @@ SUBSYSTEM_DEF(economy)
...
+	if(service)
+		service.adjust_money(MAX_GRANT_DPT)
+	if(medical)
+		medical.adjust_money(MAX_GRANT_DPT)
+	if(security)
+		security.adjust_money(MAX_GRANT_DPT)
+	if(science)
+		science.adjust_money(MAX_GRANT_DPT)
+	if(civilian)
+		civilian.adjust_money(MAX_GRANT_DPT)
```
Comment (ExcessiveUseOfCobblestone) [inline code/controllers/subsystem/economy.dm:123]:
```DM
var/list/accounts = list(ACCOUNT_1,ACCOUNT_2, blah)

for(var/iteration in accounts)
    var/dept_account = get_dep_account(iteration)
    if(!dept_account)
        continue
    dept_account.adjust_money(MAX_GRANT_DPT)
```
there might even be a pre-existing list for this sucker.

**Comment by Pugie** [comment]
Oh boy, I love having to do fetch quests for 10% cuts in order to have a chance feed my pet catgirl.

**Comment by adamkad1** [comment]
Its fine, stealing budget cards from heads lockers or peoples id's for their money is how we should play the game

**Pull request merged**